cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,348 @@
+		return err
+	}
+	return secretbootstrap.SaveConfigToFile(secretBootstrapConfigFile, &c)
+}
+
+func updateCiSecretBootstrapConfig(o options, c *secretbootstrap.Config) error {
+	for _, groupName := range []string{buildUFarm, "non_app_ci", "non_app_ci_x86"} {
+		c.ClusterGroups[groupName] = append(c.ClusterGroups[groupName], o.clusterName)
+	}
+	c.UserSecretsTargetClusters = append(c.UserSecretsTargetClusters, o.clusterName)
+	if err := updatePodScalerSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateBuildFarmSecrets(c, o); err != nil {
+		return err
+	}
+	if err := updateDPTPControllerManagerSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateRehearseSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateChatBotSecret(c, o); err != nil {
+		return err
+	}
+	if err := updateExistingRegistryPullCredentialsAllSecrets(c, o); err != nil {
+		return err
+	}
+	appendSecret(generateRegistryPushCredentialsSecret, c, o)
+	appendSecret(generateRegistryPullCredentialsSecret, c, o)
+	appendSecret(generateCiOperatorSecret, c, o)
+	generateRegistryPullCredentialsAllSecrets(c, o)
+	return nil
+}
+
+func appendSecret(secretGenerator func(options) secretbootstrap.SecretConfig, c *secretbootstrap.Config, o options) {
+	secret := secretGenerator(o)
+	logrus.Infof("Creating new secret with 'to' of: %v", secret.To)
+	c.Secrets = append(c.Secrets, secret)
+}
+
+func generateCiOperatorSecret(o options) secretbootstrap.SecretConfig {
+	return secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			kubeconfig: {
+				Field: serviceAccountKubeconfigPath(ciOperator, o.clusterName),
+				Item:  buildUFarm,
+			},
+		},
+		To: []secretbootstrap.SecretContext{
+			{
+				Cluster:   o.clusterName,
+				Name:      ciOperator,
+				Namespace: testCredentials,
+			},
+		},
+	}
+}
+
+func generateRegistryPushCredentialsSecret(o options) secretbootstrap.SecretConfig {
+	return secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			dotDockerConfigJson: generatePushPullSecretFrom(o.clusterName, []secretbootstrap.DockerConfigJSONData{
+				{
+					AuthField:   "token_image-pusher_app.ci_reg_auth_value.txt",
+					Item:        buildUFarm,
+					RegistryURL: api.ServiceDomainAPPCIRegistry,
+				},
+			}),
+		},
+		To: []secretbootstrap.SecretContext{
+			generateDockerConfigJsonSecretConfigTo(api.RegistryPushCredentialsCICentralSecret, ci, o.clusterName),
+			generateDockerConfigJsonSecretConfigTo(api.RegistryPushCredentialsCICentralSecret, testCredentials, o.clusterName),
+		},
+	}
+}
+
+func generateRegistryPullCredentialsSecret(o options) secretbootstrap.SecretConfig {
+	return secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			dotDockerConfigJson: generatePushPullSecretFrom(o.clusterName, []secretbootstrap.DockerConfigJSONData{
+				{
+					AuthField:   registryPullTokenField(ci),
+					Item:        buildUFarm,
+					RegistryURL: "registry.svc.ci.openshift.org",
```

Comment:
There's a constant for this already right?

Reply:
Not that I am seeing, but perhaps there should be.